### PR TITLE
Lisa login page

### PR DESCRIPTION
**Pull request by 2016lisali**
1. add "error", "info", "success", "warning" color variables in theme palette. 
2. add login and register page
3. refactor if else statement in contact.tsx file and refactor the test file
4. remove the alert from contact.tsx file. The undeliverable email address error will be displayed in the message box under the contact form. 
5. replace 'description' with 'content' for all post data
6. change the colours  of buttons and links, modify styles according to the theme colour variables and the height of the navbar and footer

### CODE REVIEW

**Review by QiujuYang — APPROVED** [review]
Looks very nice.

File: src/pages/myAccount/MyProfile.tsx [hunk 1]
```diff
@@ -32,8 +31,15 @@ const MyProfile = () => {
...
     box: {
-        mt: 3,
-        mb: 5,
+        display: "flex",
+        flexDirection: "column",
+        justifyContent: "center",
+        alignItems: "center",
+        py: 4,
+        height: {
+            sm: "calc(100vh - 64px - 305px)", // 100vh - header height - footer height
```
Comment (Jet004) [inline src/pages/myAccount/MyProfile.tsx:40]: As above, hard coded values can cause problems if something changes. We should find a better, more dynamic, solution.

File: src/pages/myAccount/MyProfile.tsx [hunk 2]
```diff
@@ -32,8 +31,15 @@ const MyProfile = () => {
...
+        },
+        minHeight: "630px",
```
Comment (Jet004) [inline src/pages/myAccount/MyProfile.tsx:42]: As above

**Review by Jet004 — CHANGES_REQUESTED** [review]
Nice work overall, just a few things which may become problems in the future which we should be aware of

File: src/pages/Login.tsx [hunk 3]
```diff
@@ -1,7 +1,135 @@
...
+    },
+    link: {
+        px: 1,
+    },
+    pageContainer: {
+        px: 0,
+        display: "flex",
+        justifyContent: "center",
+        alignItems: "center",
+        height: "calc(100vh - 64px - 305px)", // 100vh - header height - footer height
```
[the remark this answers is not in the document]
Comment (2016lisali) [reply src/pages/Login.tsx:113]: Changed to flexGrow. Thank you for your suggestion :)

File: src/pages/Login.tsx [hunk 4]
```diff
@@ -1,7 +1,135 @@
...
+        minHeight: "526px",
```
[the remark this answers is not in the document]
Comment (2016lisali) [reply src/pages/Login.tsx:114]: The mine-height was nolonger needed after using flexGrow. Thank you for the suggestion :).

File: src/pages/Login.tsx [hunk 5]
```diff
@@ -1,7 +1,135 @@
...
+    pwLink: {
+        fontSize: "14px",
+        py: 2,
+        textDecoration: "none",
+    },
+    signup: {
+        alignItems: "center",
+    },
+    terms: {
+        color: "grey",
```
[the remark this answers is not in the document]
Comment (2016lisali) [reply src/pages/Login.tsx:125]: The colour was replaced by theme colors. Thank you for the suggestion :)

File: src/pages/Register.tsx [hunk 6]
```diff
@@ -1,7 +1,144 @@
...
+        display: "flex",
+        alignItems: "center",
+    },
+    pageContainer: {
+        px: 0,
+        display: "flex",
+        justifyContent: "center",
+        alignItems: "center",
+        height: "calc(100vh - 64px - 305px)", // 100vh - header height - footer height
+        minHeight: "691px",
```
[the remark this answers is not in the document]
Comment (2016lisali) [reply src/pages/Register.tsx:131]: minHeight has been deleted. When using the calc height, the footer will cover the form in short height screen size, so minHeight was setting to fix this issue. minHeight is no longer needed when using flex-grow.

**Review by B08waffles — APPROVED** [review]
Look's good, I agree with Jet's feedback here, good to see your using the ? where you can, makes for a much more stable system that way, good work!.

File: src/data.tsx [hunk 7]
```diff
@@ -21,7 +21,7 @@ export const posts = [
         postId: "1",
         userId: "11",
         date: "22-07-2022",
-        description:
```
[the remark this answers is not in the document]
Comment (2016lisali) [reply src/data.tsx:24]: Add buildArray in data.ts file, and add randomiser to set random category, tags and position. Add interviewItems for interview posts.

**Review by Jet004 — APPROVED** [review]
Great work! looking really good!

**Pull request merged**